### ISSUE TITLE
`commonLabels` not applying

### ISSUE THREAD

**Issue by biasedbit** [DataDog/helm-charts#1212]
**Describe what happened:**

```yaml
commonLabels:
  foo: bar

registry: public.ecr.aws/datadog
targetSystem: linux
datadog:
  apiKey: ******
  site: us5.datadoghq.com

clusterAgent:
  admissionController:
    enabled: true
    configMode: service
    failurePolicy: Fail
    mutateUnlabelled: true
```

**Describe what you expected:**

I'd expect to see `foo=bar` _somewhere_ but I can't find it anywhere in Deployments or Pods.

**Steps to reproduce the issue:**

Following the DataDog Agent [setup steps for helm](https://docs.datadoghq.com/containers/cluster_agent/setup/?tab=helm).

**Additional environment details (Operating System, Cloud provider, etc):**

Chart version: 3.40.2
Cloud Provider: EKS 1.24

**Comment by celenechang** [comment]
Hi @biasedbit , thanks for opening the issue. I just gave this a test and I see labels defined in `commonLabels` in places like the Daemonset, Cluster Agent Deployment, and ClusterRoles. You're correct that they are not present on pods, though. (If you search `datadog.labels` in the code, that can give an idea of where it is added.) Let us know if you're still seeing unexpected behavior with this parameter.

**Comment by biasedbit** [comment]
Hey @celenechang, thanks for looking into this. If I understand you correctly, these labels are, by design, only meant to be applied to the DataDog cluster agent resources?

**Comment by celenechang** [comment]
@biasedbit Sorry for the confusion. No, it is also on the Node Agent Daemonset, RBACs for the Node Agent/Cluster Agent/Cluster Checks workers, configmaps, services, network policies, and a few other resources. I think this [link](https://github.com/search?q=repo%3ADataDog%2Fhelm-charts+datadog.labels&type=code&p=1) might provide the best idea of where they are added

**Comment by biasedbit** [comment]
@celenechang Apologies if I'm being dense but all of those sound related to the DataDog Cluster Agent. To clarify my question, are any of the non-DataDog related kubernetes resources in my cluster supposed to get tagged with these `commonLabels`, or are these labels strictly applied to all the resources created/managed by the DataDog Cluster Agent helm chart?

If the former, then everything is working as expected and we just need to clarify in [the reference values.yaml](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L17-L19) file that these labels will only affect DD Cluster Agent helm chart resources. If the latter, I'm not sure what the purpose of the labels is if they're not applied to pods. If that's the case can you elaborate on use cases (i.e. why would I want to set these labels)?

TL;DR: there's a setting in the `values.yaml` that needs clarification on purpose and use cases to avoid getting folks like myself confused 🙃

**Comment by celenechang** [comment]
Hi @biasedbit, I'll try to provide enough context to clarify the behavior, and my previous messages.

The `datadog` Helm chart does not manage non-Datadog-related resources, so it is not able to modify them.

The `datadog` Helm chart manages three main resources - the Datadog Agent, which runs as a Daemonset, the Datadog Cluster Agent (a Deployment), and Datadog Cluster Checks workers (a Deployment). All three are recommended for use in a monitoring set up with Datadog, but they are not all required. Other items like RBACs are also created/managed for these main resources to function optimally.

The values.yaml file includes `agents.additionalLabels`, `clusterAgent.additionalLabels`, and `clusterChecksRunner.additionalLabels` parameters to modify labels for each of these resources individually. I presume that a `commonLabels` parameter was added to make it easier for users to add the same labels to all three resources (and peripherals such as RBACs, configmaps, services, and so on).

Does that help?

**Comment by biasedbit** [comment]
Yep, that clarifies everything. I'd suggest making that clear in [the values.yaml](https://github.com/DataDog/helm-charts/blob/main/charts/datadog/values.yaml#L17-L19) template file.

I'm going to go ahead and close this ticket, thank you @celenechang!
